backend/src/index.ts

```diff
@@ -296,77 +306,98 @@ app.get("/user", async (req, res: Response) => {
+    }
 
-app.post("/event/create", async (req: TypedRequest<CreateEventBody>, res:Response) => {
-  //Session validation
-  const event = req.body 
+    try {
+      const newSociety = await prisma.society.create({
+        data: {
+          name: society.name,
+          admin: {
+            connect: {
+              id: sessionFromDB.userId,
+            },
+          },
+        },
+      });
 
-  const sessionFromDB = await validateSession(req.session ? req.session : null);
-  if (!sessionFromDB) {
-    return res.status(401).json({ message: 'Invalid session provided.' });
+      return res.status(200).json(newSociety);
+    } catch (e) {
+      return res.status(400).json({message:"invalid society input"});
+    }
   }
+);
 
-  //Sanitize Inputs/Check Validity
-  if(!isValidDate(event.startDateTime, event.endDateTime)) {
-    return res.status(400).json({ message: "Invalid date"})
-  }
-  
-  const eventRes = await prisma.event.create({
-    data: {
-      banner: event.banner,
-      name: event.name,
-      startDateTime: dayjs(event.startDateTime).toISOString(),
-      endDateTime: dayjs(event.endDateTime).toISOString(),
-      location: event.location,
-      description: event.description,
-      society: {
-        connect: {
-          id: event.societyId
-        }
-      }
+app.post(
+  "/event/create",
+  async (req: TypedRequest<CreateEventBody>, res: Response) => {
+    //Session validation
+    const event = req.body;
+
+    const sessionFromDB = await validateSession(
+      req.session ? req.session : null
+    );
+    if (!sessionFromDB) {
+      return res.status(401).json({ message: "Invalid session provided." });
     }
-  })
-  return res.status(200).json({ eventRes })
-})
 
-function isValidDate(startDate:Date, endDate:Date):boolean{
+    //Sanitize Inputs/Check Validity
+    if (!isValidDate(event.startDateTime, event.endDateTime)) {
+      return res.status(400).json({ message: "Invalid date" });
+    }
+
+    try {
+      const eventRes = await prisma.event.create({
+        data: {
+          banner: event.banner,
+          name: event.name,
+          startDateTime: dayjs(event.startDateTime).toISOString(),
+          endDateTime: dayjs(event.endDateTime).toISOString(),
+          location: event.location,
+          description: event.description,
+          society: {
+            connect: {
+              id: event.societyId,
+            },
+          },
+        },
+      });
+      return res.status(200).json({ eventRes });
+    } catch (e) {
+      return res.status(400).json({message:"Invalid event input"})
+    }
+  }
+);
+
+function isValidDate(startDate: Date, endDate: Date): boolean {
   //Should probably add more test cases here
-  var parsedStartDate = dayjs(startDate)
-  var parsedEndDate = dayjs(endDate)
-  //Might need to keep an eye on this condition, I'm subtracting two mins from dayjs as allowance as 
+  var parsedStartDate = dayjs(startDate);

```

Comment:
please use `const`/`let` instead of `var` throughout the program, including tests